crates/iota-metrics/src/hardware_metrics.rs

```diff
@@ -0,0 +1,402 @@
+            Ok(bytes) => bytes,
+            Err(e) => {
+                tracing::error!("Failed converting memory_available_bytes to i64: {e}");
+                return None;
+            }
+        };
+        self.memory_available_collector.set(memory_available_bytes);
+        Some(self.memory_available_collector.collect())
+    }
+
+    fn disk_has_db(disk: &Disk, db_path: &Path) -> bool {
+        db_path.starts_with(disk.mount_point())
+    }
+    fn collect_disk_available(&self, disks: &Disks) -> Vec<MetricFamily> {
+        let space_available_per_disk: Vec<MetricFamily> = disks
+            .iter()
+            .enumerate()
+            .map(|(idx, disk)| {
+                let disk_name = disk.name().to_string_lossy();
+                let disk_num = idx + 1;
+                Self::uint_gauge(
+                    &format!("disk_{disk_num}_available_bytes",),
+                    &format!("Disk available space in bytes, for disk {disk_num}",),
+                    disk.available_space(),
+                    &[
+                        Some(Self::label("disk_name", disk_name.to_string())),
+                        if Self::disk_has_db(disk, &self.db_path) {
+                            Some(Self::label("is_database_disk", true))
+                        } else {
+                            None
+                        },
+                    ],
+                )
+            })
+            .collect();
+
+        space_available_per_disk
+    }
+
+    fn collect_disks_total_bytes(disks: &Disks, db_path: &Path) -> Vec<MetricFamily> {
+        let total_bytes_per_disk: Vec<MetricFamily> = disks
+            .iter()
+            .enumerate()
+            .map(|(idx, disk)| {
+                let disk_name = disk.name().to_string_lossy();
+                let disk_num = idx + 1;
+                Self::uint_gauge(
+                    &format!("disk_{disk_num}_total_bytes",),
+                    &format!("Disk total space in bytes, for disk {disk_num}",),
+                    disk.total_space(),
+                    &[
+                        Some(Self::label("disk_name", disk_name.to_string())),
+                        if Self::disk_has_db(disk, db_path) {
+                            Some(Self::label("is_database_disk", true))
+                        } else {
+                            None
+                        },
+                    ],
+                )
+            })
+            .collect();
+
+        total_bytes_per_disk
+    }
+}
+
+impl Collector for HardwareMetrics {
+    fn desc(&self) -> Vec<&Desc> {
+        self.static_descriptions.iter().collect()
+    }
+
+    fn collect(&self) -> Vec<MetricFamily> {
+        let mut system = match self.system.lock() {
+            Ok(lock) => lock,
+            Err(e) => {
+                tracing::error!("Failed acquiring lock on System: Lock is poisoned: {e}");
+                return Vec::new();
+            }
+        };
+        system.refresh_all();
```

Comment:
But why do we refresh everything?
This will also refresh CPU etc, so this might be blocking, no? Can't we just refresh the available memory and the disks?

Reply:
Yes that's a good point, we only need memory now, I updated it to only refresh this.